api-server/routes/auth.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from httpx import AsyncClient
+from fastapi import APIRouter, status, HTTPException
+from fastapi.responses import RedirectResponse
+from urllib.parse import urlencode
+from config import (
+    GITHUB_LOGIN_ENDPOINT,
+    GITHUB_ACCESS_TOKEN_ENDPOINT,
+    GITHUB_CLIENT_ID,
+    GITHUB_CLIENT_SECRET,
+)
+from services.github_client import (
+    GithubClient,
+    GithubUserNotFound,
+    GithubUserLoadingError,
+)
+from services.auth import Auth
+
+router = APIRouter(prefix="/auth", tags=["Auth"])
+logger = logging.getLogger()
+
+
+def build_protected_response(url: str, token: str):
+    response = RedirectResponse(url)
+    # todo(rohan) : extract domain from url
+    response.set_cookie(key="token", value=token, httponly=True, domain="127.0.0.1")
+    return response
+
+
+@router.get("/login", status_code=status.HTTP_307_TEMPORARY_REDIRECT)
+async def login():
```

Comment:
What are the limits for GitHub authentications? Do GitHub authentications count as usages of GitHub's Rest API?

Reply:
I think the limit is 5000 requests/hour. We are using an OAuth App, so I think [this ](https://docs.github.com/en/apps/creating-github-apps/creating-github-apps/rate-limits-for-github-apps#user-to-server-requests)applies to us.

Reply:
Gotcha